app/Http/Controllers/Backend/MediaWikiHostController.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Http\Controllers\Backend;
+
+use App\Http\Controllers\Controller;
+use App\Services\MediaWikiHostResolver;
+use App\Services\UnknownDBVersionException;
+use App\Services\UnknownWikiDomainException;
+use Illuminate\Http\Request;
+
+
+class MediaWikiHostController extends Controller {
+    public function getWikiHostForDomain(Request $request): \Illuminate\Http\JsonResponse {
+        $mediawikiHostResolver = new MediaWikiHostResolver();
+        $domain = $request->query('domain');
+        try {
+            $backendHost =  $mediawikiHostResolver->getBackendHostForDomain($domain);
+            $webHost = $mediawikiHostResolver->getWebHostForDomain($domain);
+            $apiHost = $mediawikiHostResolver->getApiHostForDomain($domain);
```

Comment:
This now means that `MediaWikiHostResolver` will do the same DB query 3 times to return the same result each time. We should think about how to avoid this, e.g. do some caching in the `MediaWikiHostResolver` class.